src/com/opc_ua/opcua_objectstruct_helper.cpp

```diff
@@ -133,9 +136,10 @@ bool COPC_UA_ObjectStruct_Helper::addOPCUAStructTypeComponent(UA_Server *paServe
     vAttr.dataType = COPC_UA_Helper::getOPCUATypeFromAny(*paStructMember)->typeId;
 
   UA_NodeId memberNodeId;
+  char *memberBrowsePathStr = nullptr;
   if(paParentNodeId.identifierType == UA_NODEIDTYPE_STRING) {
-    std::string memberBrowsePathStr = getStructMemberBrowsePath(paStructName, paStructMemberNameId);
-    memberNodeId = UA_NODEID_STRING(mOpcuaTypeNamespaceIndex, &memberBrowsePathStr[0]);
```

Comment:
Is there a good reason not to use

`memberNodeId = UA_NODEID_STRING_ALLOC(mOpcuaTypeNamespaceIndex, memberBrowsePathStr.c_str());`

As far as I understand it that should solve this problem without having to resort to returning raw pointers and manually handling all of the memory management (The UA_NodeId_clear calls are still necessary)

Reply:
Agree. `memberNodeId` is not properly handling its memory and should copy from `memberBrowsePathStr` which is doing fine. That will also keep `getStructMemberBrowsePath` untouched.

We might want to start thinking about some wrappers around `UA_NodeId` which handle the memory safely so we don't miss this kind of errors (something like this UA_String wrapper https://github.com/eclipse-4diac/4diac-forte/pull/239/files#diff-628010b336b740e9af10dde6015a0787f2140f60a1898ce16bbc59f872d6e539R65)

Reply:
That's true. I didn't think of that. However, when I'm using something like `memberNodeId = UA_NODEID_STRING_ALLOC(mOpcuaTypeNamespaceIndex, memberBrowsePathStr.c_str());`, then the address sanitizer is reporting a memory leak.  The String allocation seems to not be correctly freed.  That is also the case when I try to clear the UA_string additionaly. 
Allocating the string with the  `UA_String_fromChars` method, assigning it manually to the NodeId and then clearing it works.

Reply:
@cochicde That is true. A wrapper might might help to avoid these issues. We should discuss that with @azoitl.

Reply:
`UA_NODEID_STRING_ALLOC` allocates a new string, and `UA_NodeId_clear` de-allocates it.

I think the problem you're facing is related to the way you're calling  `UA_Server_addVariableNode`. The last parameter is the output node Id, i.e. the node ID of the new node it was created. You're passing the parent node (`memberNodeId`) also there, meaning that's overwritten. If you are not using the result NodeId you should pass `nullptr`

Here's the snipped code that handles that when adding a new node (`memberNodeId` becomes `outNewId` when calling the function)

```
 if(outNewNodeId)
        *outNewNodeId = result.addedNodeId;
    else
        UA_NodeId_clear(&result.addedNodeId);
```

Reply:
Thanks for pointing that out. In this case, the output node Id is not needed. I replaced it with a nullptr. Now the STRING_ALLOC function works as intended.